opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -496,7 +496,7 @@ def use_span(
                     span_snapshot
                 )
         finally:
-            if end_on_exit:
+            if end_on_exit and span.end_time is None:
```

Comment:
It looks reasonable, but if someone `end` the span explicitly in scope of context manager (like we do it in tests) by mistake, it may be useful to have this warning. What do you think?

Reply:
This one is a bit more complicated. I will be very nice to have zero warnings raised in the tests. Nevertheless [this](https://github.com/open-telemetry/opentelemetry-python/pull/358/files/62b030258d3a1d38129ceb1af42cf0aac3de1b75..91b1a8cb77d8e3fb240783f19d17a18c8672ffce#diff-6cc4cd458dadfa03bce99b851b232fd2R550) particular test line makes it hard to use `self.assertLogs` just to catch the warning raised in `__exit__` without also of catching every warning raised in the `with` context. I'll think about a possible solution.

Reply:
Ok, there. I added a solution for this problem.